nni/contrib/compression/base/setting.py

```diff
@@ -0,0 +1,222 @@
+
+    @classmethod
+    def _update_setting(cls, setting1: _SETTING, setting2: _SETTING):
+        for target_name, target_setting in setting2.items():
+            if target_name in setting1:
+                setting1[target_name].update(deepcopy(target_setting))
+            else:
+                setting1[target_name] = deepcopy(target_setting)
+
+    @classmethod
+    def _update_shortcut_setting(cls, setting1: _SETTING, setting2: Dict[str, Any]):
+        for _, target_setting in setting1.items():
+            update_keys = [key for key in target_setting.keys() if key in setting2]
+            for key in update_keys:
+                target_setting[key] = deepcopy(setting2[key])
+
+
+class PruningSetting(ModuleSetting):
+    default_setting = {
+        'weight': {
+            'sparse_ratio': None,
+            'max_sparse_ratio': None,
+            'min_sparse_ratio': None,
+            'sparse_threshold': None,
+            'global_group_id': None,
+            'dependency_group_id': None,
+            'granularity': 'default',
+            'internal_metric_block': None,
+            'apply_method': 'mul',
+        },
+        'bias': {
+            # should support granularity in the future
+            # module_name should also be supported
+            'align': {
+                'target_name': 'weight',
+                'dims': [0],
+            },
+            'apply_method': 'mul',
+        }
+    }
+
+    # TODO: add more default module type
+    registry = {
+        'Linear': default_setting,
+        'Conv2d': default_setting,
+    }
+
+
+class QuantizationSetting(ModuleSetting):
+    default_setting = {
+        '_input_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        },
+        'weight': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        },
+        # TODO: in which case should bias be quantized? need to check
+        'bias': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        },
+        '_output_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        }
+    }
+
+    avtivation_setting = {
+        '_input_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        },
+        '_output_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        }
+    }
+
+    # TODO: add more default module type
+    registry = {
+        'Linear': default_setting,
```

Comment:
Is it necessary to provide an interface for the user to register the settings of the submodel? For example, the user want to prune multi-head attention layer,  there is no interface to register the settings of the multi-head attention layer  in this pipeline.

Reply:
the interface is at line 26